cartographer/mapping/3d/submap_3d.cc

```diff
@@ -332,6 +336,11 @@ void ActiveSubmaps3D::AddSubmap(
     // This will crop the finished Submap before inserting a new Submap to
     // reduce peak memory usage a bit.
     CHECK(submaps_.front()->insertion_finished());
+    // We do `ForgetIntensityHybridGrid` in order to reduce memory usage.
+    // As we use active submaps, in particular intensity hybrid grids associated
+    // with them, for scan matching, thus we call `ForgetIntensityHybridGrid`
+    // only when we are about to remove the submap from active submaps.
```

Comment:
This sentence is a bit bumpy to read. How about this:

> As we use active submaps for scan matching, in particular intensity hybrid grids associated with them, we call `ForgetIntensityHybridGrid` only when we are about to remove the submap from active submaps.

Reply:
Rephrased. What do you think?